### ISSUE TITLE
Step-by-step tutorial with image dataset

### ISSUE THREAD

**Issue by generall** [qdrant/quaterion#69]
We need to create a tutorial for users who have never used Quaterion and want to go through it step-by-step.

In the tutorial we need

- Describe what is the use-case of Quaterion - why not just use e.g. pytorch-metric-learning
- Have a proper project layout - encoders are defined in separate module, it should be possible to use them after training
- Description of what components serve what purposes
  - why we use Pytorch Lightning - what is our responsibility and what is theirs (with links to their docs)
  - what is the difference between trainable model and MetricModel
  - Why do we need a separate quaterion-models
  - How to properly initialize pre-trained encoders
